### ISSUE TITLE
Incompatibility with uvicorn

### ISSUE THREAD

**Issue by jon-maki** [bcb/jsonrpcclient#149]
The current released version of jsonrpcclient (3.3.5) is incompatible with the current released version of uvicorn (0.11.3). This is because the two libraries depend on the `click` library but required incompatible versions:

```bash
$ echo  -e "jsonrpcclient\nuvicorn" > requirements.in
$ pip-compile --verbose --generate-hashes
Using indexes:
  https://pypi.python.org/simple

                          ROUND 1
Current constraints:
  jsonrpcclient (from -r requirements.in (line 1))
  uvicorn (from -r requirements.in (line 2))

Finding the best candidates:
  found candidate jsonrpcclient==3.3.5 (constraint was <any>)
  found candidate uvicorn==0.11.3 (constraint was <any>)

Finding secondary dependencies:
  uvicorn==0.11.3           requires click==7.*, h11<0.10,>=0.8, httptools==0.1.*; sys_platform !=
"win32" and sys_platform != "cygwin" and platform_python_implementation != "PyPy", uvloop>=0.14.0;
sys_platform != "win32" and sys_platform != "cygwin" and platform_python_implementation != "PyPy",
websockets==8.*
  jsonrpcclient==3.3.5      requires apply-defaults<1, click<7, jsonschema<4

New dependencies found in this round:
  adding ['apply-defaults', '<1', '[]']
  adding ['click', '<7,==7.*', '[]']
  adding ['h11', '<0.10,>=0.8', '[]']
  adding ['httptools', '==0.1.*', '[]']
  adding ['jsonschema', '<4', '[]']
  adding ['uvloop', '>=0.14.0', '[]']
  adding ['websockets', '==8.*', '[]']
Removed dependencies in this round:
------------------------------------------------------------
Result of round 1: not stable

                          ROUND 2
Current constraints:
  apply-defaults<1 (from jsonrpcclient==3.3.5->-r requirements.in (line 1))
  click<7,==7.* (from uvicorn==0.11.3->-r requirements.in (line 2))
  h11<0.10,>=0.8 (from uvicorn==0.11.3->-r requirements.in (line 2))
  httptools==0.1.* (from uvicorn==0.11.3->-r requirements.in (line 2))
  jsonrpcclient (from -r requirements.in (line 1))
  jsonschema<4 (from jsonrpcclient==3.3.5->-r requirements.in (line 1))
  uvicorn (from -r requirements.in (line 2))
  uvloop>=0.14.0 (from uvicorn==0.11.3->-r requirements.in (line 2))
  websockets==8.* (from uvicorn==0.11.3->-r requirements.in (line 2))

Finding the best candidates:
  found candidate apply-defaults==0.1.4 (constraint was <1)
Could not find a version that matches click<7,==7.* (from uvicorn==0.11.3->-r requirements.in (line 2))
Tried: 0.1, 0.1, 0.2, 0.2, 0.3, 0.3, 0.4, 0.4, 0.5, 0.5, 0.5.1, 0.5.1, 0.6, 0.6, 0.7, 0.7, 1.0, 1.0, 1.1, 1.1, 2.0, 2.0, 2.1, 2.1, 2.2, 2.2, 2.3, 2.3, 2.4, 2.4, 2.5, 2.5, 2.6, 2.6, 3.0, 3.0, 3.1, 3.1, 3.2, 3.2, 3.3, 3.3, 4.0, 4.0, 4.1, 4.1, 5.0, 5.0, 5.1, 5.1, 6.0, 6.0, 6.1, 6.1, 6.2, 6.2, 6.3, 6.3, 6.4, 6.4, 6.5, 6.6, 6.6, 6.7, 6.7, 7.0, 7.0, 7.1, 7.1, 7.1.1, 7.1.1
Skipped pre-versions: 6.7.dev0
There are incompatible versions in the resolved dependencies:
  click<7 (from jsonrpcclient==3.3.5->-r requirements.in (line 1))
  click==7.* (from uvicorn==0.11.3->-r requirements.in (line 2))
```

The current `master` branch of jsonrpcclient has this dependency fix, as of commit https://github.com/bcb/jsonrpcclient/commit/8ce1bbaa1db1ccda96d38f316d881aef0667c1fa.

My current workaround for this is to `pip3` install directly from github, but would love to see a point release of jsonrpcclient that resolves this dependency mismatch.

**Comment by lanquil** [comment]
The author already declared his intention to do so in https://github.com/bcb/jsonrpcclient/pull/147#issuecomment-612113546

**Comment by bcb** [comment]
Released 3.3.6, should be fixed now.

**Comment by jon-maki** [comment]
Works great, thank you!
